esp-hal/src/i2c/master/mod.rs

```diff
@@ -85,6 +86,107 @@ const I2C_CHUNK_SIZE: usize = 254;
 // on ESP32 there is a chance to get trapped in `wait_for_completion` forever
 const MAX_ITERATIONS: u32 = 1_000_000;
 
+/// Address mode.
+///
+/// Note: This trait is sealed and should not be implemented outside of this
+/// crate.
+pub trait AddressMode: _private::AddressModeInternal + Copy + 'static {}
+
+mod _private {
+    pub trait AddressModeInternal {
```

Comment:
I don't understand the need for this. I2cAddress can `impl From` from the possible address types.

Reply:
idea is that implementing `From` extends the public API surface and we would need to expose the internal enum

Reply:
Why is that a problem, since that enum is already marked with a bunch of derives that only make sense if it's public?

Reply:
we can ofc remove the unnecessary derives - we just shouldn't make anything public which doesn't need to